qiskit_ibm_runtime/ibm_runtime_service.py

```diff
@@ -171,6 +171,7 @@ def __init__(
         )
         self._backends: Dict[str, "ibm_backend.IBMBackend"] = {}
         self._api_client = None
+        self._programs_first_call = True
```

Comment:
Do we need this additional variable? If self._programs is empty doesn’t that mean it’s the first call?